File: 0189-rotate-array/0189-rotate-array.py
class Solution(object):
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        n=len(nums)
        k=k%n
        temp=[]
        
        for i in range(n-k,n):
            temp.append(nums[i])
        for i in range(n-k-1,-1,-1):
            nums[i+k]=nums[i]
        for i in range(0,k):
            nums[i]=temp[i]
        return nums

File: 0189-rotate-array/0189-rotate-array.java
class Solution {
    public void reverseNums(int[] nums,int start,int end){
        while(start<end){
            int temp=nums[start];
            nums[start]=nums[end];
            nums[end]=temp;
            start++;end--;
        }
    }
    public void rotate(int[] nums, int k) {
        int n=nums.length;
        k=k%(n);
        reverseNums(nums,n-k,n-1);
        reverseNums(nums,0,n-k-1);
        reverseNums(nums,0,n-1);
        

    }
}